File: codereview/codereview.js
(function () {
	'use strict';

	var codeReviewApp = angular.module('codeReviewApp');

	codeReviewApp.factory('fooService', function ($http) {
		return {
			getAllFoos: function () {
				return $http.get('/foos');
			}
		};
	});

	codeReviewApp.factory("barService", ["$http", function ($http) {
		return {
			getAllBars: function () {
				return $http.get("/bars");
			}
		}
	}]);

	codeReviewApp.controller('fooCtrl', function ($scope, fooService) {
		$scope.foos = [];

		fooService.getAllFoos().then(function (response) {
			$scope.foos = response.data.foos;
			$scope.$digest();
		});
	});

	codeReviewApp.controller("barCtrl", ["$scope", "barService", function (scope, bars) {
		scope.bars = [];

		bars.getAllBars().then(function (response) {
			scope.bars = response.data.bars;
			scope.$digest();
		});
	}]);
}());

File: codereview/codereview.cs
using System.Collections.Generic;
using System.Web;
using System.Web.Mvc;
using NewtonSoft.Json;

namespace CodeReviews
{
	public class CodeReviewController : Controller
	{
		public CodeReviewController(IFooService fooService, IBarService barService)
		{
			m_fooService = fooService;
			m_barService = barService;
		}

		public ActionResult ListFoos()
		{
			IEnumerable<FooViewModel> foos = m_fooService.GetAllFoos().Select(FooMapper.Map);
			return new JsonDotNetResult(new { foos });
		}

		public ActionResult BarService()
		{
			IEnumerable<BarViewModel> bars = m_barService.GetAllBars().Select(BarMapper.Map);
			return Json(bars);
		}

		readonly IFooService m_fooService;
		readonly IBarService m_barService;
	}
}
